zombies.js

```diff
@@ -79,250 +93,381 @@
- * -----------------------------
- * Creates a normal zombie.
- *
- * @name Zombie
- * @param {number} health           The zombie's health.
- * @param {number} strength         The zombie's strength.
- * @param {number} speed            The zombie's speed.
- * @private {number} maxHealth      Default value should be set to `health`.
- * @property {number} health
- * @property {number} strength
- * @property {number} speed
- * @property {boolean} isAlive      Default value should be `true`.
- */
+* Class => FastZombie(health, strength, speed)
+* -----------------------------
+* Creates a fast zombie.
+*
+* The FastZombie class constructor will call
+*   the super class (Zombie) constructor
+*   while passing in the 3 Zombie constructor params
+*
+* @name FastZombie
+* @param {number} health           The zombie's health.
+* @param {number} strength         The zombie's strength.
+* @param {number} speed            The zombie's speed.
+*/
 
 
 /**
- * Class => FastZombie(health, strength, speed)
- * -----------------------------
- * Creates a fast zombie.
- *
- * The FastZombie class constructor will call
- *   the super class (Zombie) constructor
- *   while passing in the 3 Zombie constructor params
- *
- * @name FastZombie
- * @param {number} health           The zombie's health.
- * @param {number} strength         The zombie's strength.
- * @param {number} speed            The zombie's speed.
- */
-
+* FastZombie Extends Zombie Class
+* -----------------------------
+*/
+
+class FastZombie extends Zombie {
+  constructor(health, strength, speed) {
+    super(Zombie);
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+  }
+}
 
 /**
- * FastZombie Extends Zombie Class
- * -----------------------------
- */
-
+* Class => StrongZombie(health, strength, speed)
+* -----------------------------
+* Creates a strong zombie.
+*
+* The StrongZombie class constructor will call
+*   the super class (Zombie) constructor
+*   while passing in the 3 Zombie constructor params
+*
+* @name StrongZombie
+* @param {number} health           The zombie's health.
+* @param {number} strength         The zombie's strength.
+* @param {number} speed            The zombie's speed.
+*/
 
 
-/**
- * Class => StrongZombie(health, strength, speed)
- * -----------------------------
- * Creates a strong zombie.
- *
- * The StrongZombie class constructor will call
- *   the super class (Zombie) constructor
- *   while passing in the 3 Zombie constructor params
- *
- * @name StrongZombie
- * @param {number} health           The zombie's health.
- * @param {number} strength         The zombie's strength.
- * @param {number} speed            The zombie's speed.
- */
-
 
 /**
- * StrongZombie Extends Zombie Class
- * -----------------------------
- */
-
-
-
+* StrongZombie Extends Zombie Class
+* -----------------------------
+*/
+
+class StrongZombie extends Zombie {
+  constructor(health, strength, speed) {
+    super(Zombie);
```

Comment:
`super` is like calling the constructor of the Parent. So passing in zombie doesn't exactly line up with the constructor arguments here.